client/login/magic-login/gravatar/index.tsx

```diff
@@ -0,0 +1,1041 @@
+									emailAddress:
+										isSecondaryEmail && ! isNewAccount ? maskedEmailAddress : usernameOrEmail,
+								},
+							}
+						) }
+					</span>
+					{ isSecondaryEmail && isNewAccount && (
+						<span>{ translate( ' A new Gravatar account will be created.' ) }</span>
+					) }
+					{ isSecondaryEmail && ! isNewAccount && (
+						<span>{ translate( ' This email already exists and is synced with Gravatar.' ) }</span>
+					) }
+				</p>
+				{ isNewAccount && <GravPoweredMagicLoginTos /> }
+				<form
+					className="grav-powered-magic-login__verification-code-form"
+					onSubmit={ handleGravPoweredCodeSubmit }
+				>
+					<FormLabel htmlFor="verification-code" hidden>
+						{ translate( 'Enter the verification code' ) }
+					</FormLabel>
+					<FormTextInput
+						id="verification-code"
+						value={ verificationCodeInputValue }
+						onChange={ handleGravPoweredCodeInputChange }
+						placeholder={ translate( 'Verification code' ) }
+						disabled={ isProcessingCode }
+						isError={ !! codeValidationError }
+						autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+					/>
+					{ codeValidationError && (
+						<Notice
+							text={ errorText }
+							className="magic-login__request-login-email-form-notice"
+							showDismiss={ false }
+							status="is-transparent-info"
+						/>
+					) }
+					<FormButton
+						primary
+						disabled={
+							! verificationCodeInputValue ||
+							verificationCodeInputValue.length < 6 ||
+							isProcessingCode
+						}
+						busy={ isProcessingCode }
+					>
+						{ translate( 'Continue' ) }
+					</FormButton>
+				</form>
+				<footer
+					className={ clsx( 'grav-powered-magic-login__footer', {
+						'grav-powered-magic-login__footer--vertical': ! isFromGravatar3rdPartyApp,
+					} ) }
+				>
+					<button
+						onClick={ () => {
+							handleGravPoweredEmailCodeSend( usernameOrEmail );
+
+							recordTracksEvent( 'calypso_gravatar_powered_magic_login_click_resend_email', {
+								type: 'code',
+								client_id: oauth2ClientId,
+								client_name: oauth2Client?.title,
+							} );
+						} }
+						disabled={ Boolean( isRequestingEmail ) || Boolean( resendEmailCountdown ) }
+					>
+						{ resendEmailCountdown === 0
+							? translate( 'Send again' )
+							: translate( 'Send again (%(countdown)d)', {
+									args: { countdown: resendEmailCountdown },
+							  } ) }
+					</button>
+					{ shouldShowSwitchEmail && (
+						<button
+							onClick={ () => {
+								resetResendEmailCountdown();
+								handleGravPoweredEmailSwitch();
+							} }
+						>
+							{ translate( 'Switch email' ) }
+						</button>
+					) }
+					<a href="https://gravatar.com/support" target="_blank" rel="noreferrer">
+						{ translate( 'Need help logging in?' ) }
+					</a>
+				</footer>
+			</div>
+		);
+	};
+
+	const GravPoweredEmailLinkVerification = () => {
+		const emailAddress =
+			usernameOrEmail && usernameOrEmail.includes( '@' ) ? usernameOrEmail : null;
```

Comment:
Seems `usernameOrEmail` is a string, we could simplify the code a little bit:

```suggestion
		const emailAddress = usernameOrEmail.includes( '@' ) ? usernameOrEmail : null;
```